php/util/class-fieldmanager-util-gutenberg.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Class file for Fieldmanager_Util_Gutenberg
+ *
+ * @package Fieldmanager
+ */
+
+/**
+ * Gutenberg helpers.
+ */
+class Fieldmanager_Util_Gutenberg {
+	public function __construct() {
+		add_filter( 'fm_enqueue_scripts', [ $this, 'add_gutenberg_js_deps' ], 99 );
+
+		// Gutenberg sidebar polyfill.
+		fm_add_script( 'fieldmanager-gutenberg-polyfill', 'js/fieldmanager-gutenberg-support.js' );
+	}
+
+	/**
+	 * Shim for fields loaded inside of Gutenberg editor.
+	 *
+	 * Adds JS dependency to all fm scripts to ensure proper load order.
+	 *
+	 * @param array $scripts
+	 * @return array
+	 */
+	public function add_gutenberg_js_deps( $scripts ) {
+		$is_gutenberg_editor = false;
+
+		 // Do we have access to current screen?
+		if ( did_action( 'current_screen' ) ) {
+			$current_screen = get_current_screen();
+			$is_gutenberg_editor = $current_screen instanceof WP_Screen ? $current_screen->is_block_editor : false;
+		}
+
+		// Fallback if we don't have access to `current_screen`.
+		if ( ! $is_gutenberg_editor ) {
+			$post_id = false !== get_the_ID() ? get_the_ID() : $GLOBALS['post_ID'];
```

Comment:
- The use of a template tag seems unexpected here. What is the desired value? The ID of the post being edited?
- I think `$GLOBALS['post_ID']` will exist only only in `$_POST`. For `$_GET` requests, the ID is in `$_GET['post']`. So, there is a possibility of undefined-index warnings, and I'm not sure whether the logic here captures all the needed cases.
- Also, any value from a superglobal will need sanitization.
- Having said all that: I think `fm_enqueue_scripts` will also run on the frontend if any Fieldmanager assets are enqueued (via `\Fieldmanager_Util_Assets::hook_enqueue()`). If so, that could lead to cases where `current_screen` does not fire, but because the global post from `get_the_ID()` supports the block editor, `wp-post-edit` will be enqueued. That might be the intention here — maybe it's why `get_the_ID()` is used?

Reply:
> but because the global post from get_the_ID() supports the block editor, wp-post-edit will be enqueued

Pardon me: Not enqueued, but added as a dependency. However, if a script dependency is not enqueued, is the dependency enqueued, or does the script fail to load?